Update Installation Docs - Mainnet

BowTiedDevOps opened the pull request:
Updated docs and scripts to install bitcoin v25.0 and stacks-blockchain v2.4.0.1.0, for mainnet.
Mirroring https://github.com/stacksfoundation/miner-docs/pull/2, which is a PR for testnet.

BowTiedDevOps committed:
feat: update docs to build bitcoin 25.0 and latest stacks

wileyj commented on bitcoin.md:
```diff
@@ -29,18 +29,26 @@ _later versions of bitcoin use a named wallet, as of stacks version `2.4.0.0.0`
+    --enable-cxx \
+    --enable-static \
+    --disable-shared \
+    --prefix=/usr/local
```
```suggestion
    --bindir=/usr/local/bin
```

BowTiedDevOps committed:
feat: remove `prefix` in favor of `bindir`

wileyj commented on scripts/install_stacks.sh:
```diff
@@ -50,20 +49,44 @@ BTC_ADDRESS=$(sudo cat  /root/keychain.json | jq .keyInfo.btcAddress | tr -d '"'
     echo "[ install_stacks.sh ] - Sleeping for 120 seconds to allow bitcoin time to restart"
     sleep 120
 
+    echo "[ install_stacks.sh ] - Loading bitcoin wallet"
+    bitcoin-cli \
+        -rpcconnect=127.0.0.1 \
+        -rpcport=8332 \
+        -rpcuser=btcuser \
+        -rpcpassword=btcpass \
+        loadwallet miner
```
no longer required
```suggestion
```

BowTiedDevOps replied:
Did you manage to test it without? I've added it because after restarting the bitcoin node, the wallet is not automatically imported (perhaps we can add that feature and remove this code, but only removing it might not work). Last time I tested, I could not import the address because the wallet wasn't loaded, that's the reason of this addition.

Edit: same for the testnet pr

wileyj commented on wallet.md:
```diff
@@ -37,17 +37,36 @@ _Import will only be successful after bitcoin has fully synced_
   -rpcuser=btcuser \
   -rpcpassword=btcpass \
-getaddressinfo <npx btcAddress>
+  loadwallet miner
```
command starting on line 52 no longer required

wileyj commented:
sharing here from our discord chat:
`importmulti` will work to import an address, however if you're importing an older bitcoin address the timestamp will need to be adjusted or you won't load the address utxos. 

since the docs show how to create a new address, i think a note is a good compromise, i.e.
```
if you're importing an old address, consider changing the timestamp in the importmulti command, or use importprivkey 
<links to cli docs>
```

BowTiedDevOps committed:
feat: remove bitcoin restart and loadwallet calls

- add bitcoin wallet import on startup option in config
- option to import and use existing btc address in bitcoin wallet

BowTiedDevOps committed:
feat: remove wallet_dir from config

wileyj reviewed (approved):
👍
